src/instance.ts

```diff
@@ -758,14 +758,38 @@ Please use the format 'my-instance' or '${bigtable.projectName}/instances/my-ins
       appProfile.metadata = chunk;
       callback(null, appProfile);
     };
+    let failedLocations: string[] = [];
+    const flush = (callback: Function) => {
+      if (failedLocations.length > 0) {
+        callback(
+          new Error(
+            `Resources from the following locations are currently not available\n${JSON.stringify(
+              failedLocations
+            )}`
+          )
+        );
+      } else {
+        callback();
+      }
+    };
+    const stream = this.bigtable.request({
+      client: 'BigtableInstanceAdminClient',
+      method: 'listAppProfilesStream',
+      reqOpts,
+      gaxOpts,
+    });
+    stream.on('response', apiResp => {
+      if (apiResp.failedLocations.length > 0) {
```

Comment:
Will `failedLocations` always be defined as an array on the API response?

Reply:
As per the documentation [here](https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#listappprofilesresponse) `failed_locations` property is marked as an array.
Should I `arrify` it just in case?

Reply:
Hmm, wouldn't hurt. Does it make sense to check a real response real quick to see what happens when there aren't any failures?

Reply:
Is this what you are asking for?
![image](https://user-images.githubusercontent.com/7764119/92153034-a5a24d80-edf1-11ea-9aac-f77a15464ad6.png)

Reply:
I was thinking of actually making the request with the code and seeing what comes back. But if you’re confident from the types, don’t worry about it.

Reply:
The image is from a breakpoint of an actual [system-test](https://github.com/googleapis/nodejs-bigtable/blob/master/system-test/bigtable.ts#L166), this is how a response without `faledLocations` looked like.

Reply:
Oh, sorry. Fancy! Thanks :)